feat: Added video and updated Grafana Agent -> Alloy

Jayclifford345 opened the pull request:
Added Ingesting basics for Loki video to ` _index.md` of the Send Data directory.  Also Updated Grafana Agent to Grafana Alloy.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

JStickler commented:
@Jayclifford345 could you edit this PR so that it is just adding the video?  I've already got a WIP PR for the Alloy changes (and we do not want to delete the Agent references, as the Agent is still supported).

Jayclifford345 commented:
Hey @JStickler! Absolutely, sorry I had put in the request before our conversation on Slack. I reverted my change and just added the video. Let me know if this has changed correctly.

Jayclifford345 commented:
I also got a weird test fail I haven't seem before: `check / check / testPackages (pkg/ruler) (pull_request)`
```bash
 ✖  pkg/ruler/base (15.635s) (coverage: 32.0% of statements)

=== Failed
=== FAIL: pkg/ruler/base TestRuler_PrometheusAlerts (0.00s)
level=info msg="ruler up and running"
    testing.go:1[23](https://github.com/grafana/loki/actions/runs/9265942171/job/25489088416?pr=13032#step:5:24)1: TempDir RemoveAll cleanup: unlinkat /tmp/TestRuler_PrometheusAlerts314236595/001: directory not empty

DONE 283 tests, 1 failure in 94.108s
Error: Process completed with exit code 1.
```
